File: lib/model/src/main/kotlin/dk/example/feedback/model/error/ApiError.kt
package dk.example.feedback.model.error

import java.time.OffsetDateTime

data class ApiError(
    val timestamp: OffsetDateTime,
    val message: String,
    val domainCode: DomainCode?,
    val exceptionType: String,
    val path: String,
)

enum class DomainCode {
    FEEDBACK_ALREADY_SUBMITTED,
    EVENT_ALREADY_JOINED,
    CANNOT_JOIN_OWN_EVENT,
    CANNOT_GIVE_FEEDBACK_TO_SELF,
    PINCODE_NOT_FOUND,
}

File: ios-app/Modules/Sources/Domain/Errors/ApiError.swift
import Foundation

public struct ApiError: Error, Sendable {
    let timestamp: String?
    let message: String?
    let domainCode: DomainCode?
    let exceptionType: String?
    let path: String?
    public init(
        timestamp: String?,
        message: String?,
        domainCode: DomainCode?,
        exceptionType: String?,
        path: String?
    ) {
        self.timestamp = timestamp
        self.message = message
        self.domainCode = domainCode
        self.exceptionType = exceptionType
        self.path = path
    }
}

public enum DomainCode: Sendable {
    case feedbackAlreadySubmitted
    case eventAlreadyJoined
    case cannotJoinOwnEvent
    case cannotGiveFeedbackToSelf
    case pincodeNotFound
}
